tools/bpf2c/bpf_code_generator.cpp

```diff
@@ -529,6 +529,11 @@ bpf_code_generator::parse_btf_maps_section(const unsafe_string& name)
                 // Compute the offset of the values array and resize the vector
                 // to hold the initial values.
                 if (member.name == "values") {
+                    // If the map is statically initialized, then the keys must be uint32_t.
```

Comment:
why only 4 bytes? As long as it is integer, it should be good right?

Reply:
This is a restriction imposed by Linux. We do it this way to be compatible.